concepts/05 UI Components/DataGrid/30 Filtering and Searching/6 API/5 Filtering Columns Bound to Collection/20 Server-Side Filtering.md

```diff
@@ -0,0 +1,163 @@
+            // ...
+            <DxColumn
+                data-field="Products"
+                data-type="string"
+                :calculate-filter-expression="calculateFilterExpression"
+                :calculate-cell-value="calculateCellValue"
+            >
+            </DxColumn>
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxDataGrid, {
+        DxColumn,
+        DxRemoteOperations,
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn,
+            DxRemoteOperations,
+        },
+        data() {
+            // ...
+        },
+        methods: {
+            calculateFilterExpression(filterValue, selectedFilterOperation, target) {
+                return [
+                    [new DevExpress.data.EdmLiteral("Products/any(o: substringof('" + filterValue + "',o/ProductName) eq true)"), "=", true]
+                ];
+            },
+            calculateCellValue(data) {
+                var res = [];
+                for (var i = 0; i < data.Products.length; i++)
+                    res.push(data.Products[i].ProductName);
+                return res.join(', ');
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid, {
+        Column, 
+    } from 'devextreme-react/data-grid';
+
+    const calculateFilterExpression = (filterValue, selectedFilterOperation, target) => {
+        return [
+            [new DevExpress.data.EdmLiteral("Products/any(o: substringof('" + filterValue + "',o/ProductName) eq true)"), "=", true]
+        ];
+    }
+
+    const calculateCellValue = (data) => {
+        var res = [];
+        for (var i = 0; i < data.Products.length; i++)
+            res.push(data.Products[i].ProductName);
+        return res.join(', ');
+    }
+
+    export default function App() { 
+        return (
+            <DataGrid ... >
+                // ...
+                <Column
+                    dataField="Products"
+                    dataType="string"
+                    calculateFilterExpression={calculateFilterExpression}
+                    calculateCellValue={calculateCellValue}
+                />
+            </DataGrid>
+        );
+    }
+
+---
+
+You need to specify the [calculateFilterExpression](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateFilterExpression) property and put the `"Products/any(o: substringof("filterValue",o/ProductName) eq true)"` expression to the [EdmLiteral](/Documentation/ApiReference/Data_Layer/EdmLiteral/) method. Also, handle the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) property and create a comma-separated string from ProductNames. This string will be displayed in column cells.
```

Comment:
What do you mean by "put the X expression to the Y method"? Pass the expression to the method, am I right?

Second. This "also" does not make a proper connection between two sentences in this paragraph.

```suggestion
You need to specify the [calculateFilterExpression](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateFilterExpression) property and pass the `"Products/any(o: substringof("filterValue",o/ProductName) eq true)"` expression to the [EdmLiteral](/Documentation/ApiReference/Data_Layer/EdmLiteral/) method. In addition, you need to handle the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) property and create a comma-separated string from ProductNames. This string will be displayed in column cells.
```